app/models/recipe.rb

```diff
@@ -1,3 +1,13 @@
 class Recipe < ApplicationRecord
-   validates_presence_of :name, :complexity, :genre
+  has_many :recipe_ingredients
+  has_many :ingredients, through: :recipe_ingredients
+  validates_presence_of :name, :complexity, :genre
+
+  def self.list_ingredients
+    Ingredient.where("#{recipe.id}" = )
```

Comment:
I think this method got you tripped up a bit, I can see what you were thinking though. Let's approach the user stor again: 
```
...and I see a list of the names of the ingredients for the recipe.
```

We start with one singular recipe - does that sound like a "class" or "all the recipeS", or just one "recipe's ingredients"? If we need ALL the ingredients regardless of recipe, then I would agree with your approach of using a class method - those are great for querying the entire table. However, we just want ONE recipe's ingredients, which to me sounds like _the associated `:ingredients` for this recipe_ - an instance method. Here's what that method could look like: 

```ruby
def list_ingredients # instance method, not class
   ingredients
end
```

All that method returns is the `:ingredients` association, so we don't really need an entire method that for that really. We could also just call `@recipe.ingredients` in the view, or the controller and assign it to a different instance variable.

Reply:
If you did want to make this a class method, you'd have to send in the `recipe_id` from the controller or wherever you're calling it from, in order to set up the `where` clause correctly. But if we're looking at the context of a single recipe, you're better off using an instance method and using the association anyway.